servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/BaseGrpcClientBuilder.java

```diff
@@ -73,6 +74,14 @@
      */
     BaseGrpcClientBuilder<U, R> enableWireLogging(String loggerName);
 
+    /**
+     * Sets a {@link TransportObserver} that provides visibility into transport events.
+     *
+     * @param transportObserver A {@link TransportObserver} that provides visibility into transport events.
+     * @return {@code this}.
+     */
+    BaseGrpcClientBuilder<U, R> transportObserver(TransportObserver transportObserver);
```

Comment:
Started with the client builder approach for now. Will investigate impact of moving to `ConnectionFactory` in a follow-up PR.

Reply:
That will give more flexibility for users to correlate events/state before and after `ConnectionFactory#newConnection` invocation. Otherwise, it's hard to connect with events on the builder. Similar limitations that we discussed for `DnsQueryLifecycleObserver` in netty.
In addition, it gives users control when they need to create a new observer and for which resolved addresses.